pkg/api/validation/validation.go

```diff
@@ -4016,6 +4029,16 @@ func ValidateResourceRequirements(requirements *api.ResourceRequirements, fldPat
 		} else if resourceName == api.ResourceNvidiaGPU {
 			allErrs = append(allErrs, field.Invalid(reqPath, quantity.String(), fmt.Sprintf("must be equal to %s request", api.ResourceNvidiaGPU)))
 		}
+		if helper.IsHugePageResourceName(resourceName) {
+			reqContainsHugePages = true
+		}
+		if supportedQoSComputeResources.Has(string(resourceName)) {
+			reqContainsCpuOrMemory = true
+		}
+
+	}
+	if !limContainsCpuOrMemory && !reqContainsCpuOrMemory && (reqContainsHugePages || limContainsHugePages) {
```

Comment:
what if I don't specify the cpu, memory and hugepages in the yaml file, would this be error?

Reply:
if we do not specify the cpu, memory and hugepages , reqContainsHugePages and limContainsHugePages are both false , so the if condition is false .